### PR TITLE
Look for manual files under derivatives/

### LINKED ISSUE #3: Look for manual labels under derivatives/ folder
To follow BIDS philosophy.
Currently processing is run within the raw data folder.

### PR DESCRIPTION
**Pull request by jcohenadad**
Currently, users have to put all manual seg and label under a flat directory, under a folder called with the flag `PATH_SEGMANUAL` upon calling `sct_run_batch`.  Issues are:
- does not follow BIDS convention
- need to manually package the manual corrections and upload them for each release
- more complicated UX and documentation

This PR introduces the following changes:
- manual corrections go inside the BIDS dataset, under a `derivatives/` folder
- `process_data.sh` is modified to look for the manual corrections there

Fixes #3

Remaining TODOs (after merging this PR):
- update documentation (related to https://github.com/spine-generic/spine-generic/pull/103)
- upload manual corrections under derivatives/: https://github.com/spine-generic/spine-generic/issues/105

### CODE REVIEW

File: processing/process_data.sh [hunk 1]
```diff
@@ -78,7 +78,8 @@ segment_if_does_not_exist(){
   local contrast="$2"
   # Update global variable with segmentation file name
   FILESEG="${file}_seg"
-  FILESEGMANUAL="${PATH_SEGMANUAL}/${FILESEG}-manual.nii.gz"
+  FILESEGMANUAL="${PATH_DATA}/derivatives/${SUBJECT}/anat/${FILESEG}-manual.nii.gz"
```
Comment (valosekj) [inline processing/process_data.sh:81]: Shouldn't path change based on anat or dwi?, i.e. 
${PATH_DATA}/derivatives/${SUBJECT}/`anat`/${FILELABEL}-manual.nii.gz
or
${PATH_DATA}/derivatives/${SUBJECT}/`dwi`/${FILELABEL}-manual.nii.gz

Comment (jcohenadad) [reply processing/process_data.sh:81]: ah! great catch @valosekj. thanks! will fix right away

Comment (jcohenadad) [reply processing/process_data.sh]: Fixed in 1e07ffd

Comment (jcohenadad) [reply processing/process_data.sh]: actually, this one is better: f8021a99cb3bc1b9d2294d2294bcdac7d09ee687

**Comment by valosekj** [comment]
here https://github.com/spine-generic/spine-generic/pull/104/commits/d970b08466bc8706643c9ab3d1d4dcca0c211d9f I only changed single square bracket `[` in if-statements to double square bracket `[[` to follow https://github.com/neuropoly/spinalcordtoolbox/wiki/variables#if-statements

**Pull request merged**